src/hooks/screen/useCreatePostHook.js

```diff
@@ -42,20 +37,21 @@
     getAnonUserInfo();
   }, []);
 
-  React.useEffect(() => {
-    getAnonUserInfo();
-  }, []);
-
   React.useEffect(() => {
     if (!isAnonymous) getAnonUserInfo();
   }, [isAnonymous]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This `useEffect` hook will call `getAnonUserInfo` every time `isAnonymous` changes. However, it only makes sense to fetch anonymous user info if `isAnonymous` is true. Consider adding a condition inside the `useEffect` to check if `isAnonymous` is true before calling `getAnonUserInfo`.

```diff
- if (!isAnonymous) getAnonUserInfo();
+ if (isAnonymous) getAnonUserInfo();
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->